File: OOP/1_intro/Reckless+Driver+-+Java/Reckless Driver/src/com/poash/rd/Sedan.java
package com.poash.rd;

public class Sedan extends Vehicle {
	public Sedan(int damage, int cash){
		super(damage, cash) ;
		setName("Sedan");
	}
	@Override
	public void OnCollision(GameObject other) {
		if(other.getTag().equals("Player")){
			Player player = (Player)other; 
				System.out.println("#### COLLISION ->[Sedan] Sparks flying");
				player.ApplyDamage(getDamage(), getCash());
		}
	}
}

File: OOP/1_intro/Reckless+Driver+-+C++/RecklessDriver/RecklessDriver/Sedan.cpp
#include "Sedan.h"
#include "Player.h"
#include <iostream>
namespace RecklessDriver {

	Sedan::Sedan(int _damage, int _cash):Vehicle(_damage, _cash)
	{
		SetName("Sedan");
	}

	void Sedan::OnCollision(GameObject *other)
	{
		if (other->GetTag() == "Player") {
			Player *p = (Player *)other;
				std::cout << "#### COLLISION ->[Sedan] Sparks flying" << std::endl;
				p->ApplyDamage(GetDamage(), GetCash());
		}
	}
}
